File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.cpp
class Solution {
public:
    int countCharacters(vector<string>& words, string chars) {
        int result = 0;
        for (const string& word : words) {
            string tempChars = chars;
            bool valid = true;
            for (char c : word) {
                size_t pos = tempChars.find(c);
                if (pos != string::npos) {
                    tempChars.erase(pos, 1);
                } else {
                    valid = false;
                    break;
                }
            }
            if (valid) {
                result += word.size();
            }
        }
        return result;
    }
};

File: 1160-find-words-that-can-be-formed-by-characters/1160-find-words-that-can-be-formed-by-characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        def can_form(word, chars):
            char_count = Counter(chars)
            for char in word:
                if char_count[char] <= 0:
                    return False
                char_count[char] -= 1
            return True
        result = 0
        for word in words:
            if can_form(word, chars):
                result += len(word)
        return result